server-side/node-mssql-crud/src/database/query.js

```diff
@@ -93,6 +93,7 @@ export const queries = {
 
   acceptGatepass: "UPDATE [gps_db].[gps_db].[gps_gatepassmaster] SET status='Approved' WHERE request_id=@id;",
   rejectGatepass: "UPDATE [gps_db].[gps_db].[gps_gatepassmaster] SET status='Rejected' WHERE request_id=@id;",
+  cancelApprovedGatepass:"UPDATE [gps_db].[gps_gatepassmaster] SET status = 'cancelled', approved_or_rejected_by= @user_id , approved_or_rejected_date = @date, approved_or_rejected_time = @time, comments = @comments WHERE request_id= @req_id",
```

Comment:
1. do we update approved_or_rejected_date / time when we cancel the gatepass?? I think it is only updated when gatepass is approved or cancelled. same for approved_or_rejected_by. Please tell.